code/ai/aicode.cpp

```diff
@@ -15541,7 +15553,7 @@ void ai_ship_hit(object *objp_ship, object *hit_objp, vec3d *hit_normal)
 		objp_hitter = &Objects[hitter_objnum];
 
 		// lets not check hits by ghosts any further either
-		if(objp_hitter->type == OBJ_GHOST)
+		if(objp_hitter->type != OBJ_SHIP && objp_hitter->type != OBJ_WEAPON && objp_hitter->type != OBJ_ASTEROID && objp_hitter->type != OBJ_BEAM)
```

Comment:
`OBJ_DEBRIS` is also a possibility.  But I would add an assert here as nothing that isn't those object types has any business getting to this point.

Reply:
Ok, those have been added.